nupic/research/frameworks/dynamic_sparse/networks/layers.py

```diff
@@ -717,59 +910,84 @@ class RandDSConv2d(DSConv2d):
     random.
     """
 
-    def progress_connections(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.coactivations.data[:] = torch.ones_like(self.weight)
 
-        if self.prune_grads_hook is not None:
-            self.prune_grads_hook.remove()
+    # def progress_connections(self, *args, **kwargs):
+
+    #     if self.prune_grads_hook is not None:
+    #         self.prune_grads_hook.remove()
+
+    #     with torch.no_grad():
+
+    #         keep_mask = torch.rand(self.weight.shape) < self.nonzero_frac
+    #         self.weight[~keep_mask] = 0
+    #         self.prune_grads_hook = self.weight.register_hook(
+    #             lambda grad: grad * keep_mask.type(grad.dtype).to(grad.device))
+    #         self.pruning_iterations += 1
+
+    #         if self.last_keep_mask is not None:
+    #             kept = (self.last_keep_mask == keep_mask)
+    #             kept = kept[keep_mask == 1]
+    #             self.kept_frac = 1 - calc_sparsity(kept)
+    #         self.last_keep_mask = keep_mask
+
+    def update_coactivations(self, input_tensor, output_tensor):
 
         with torch.no_grad():
 
-            keep_mask = torch.rand(self.weight.shape) < self.nonzero_frac
-            self.weight[~keep_mask] = 0
-            self.prune_grads_hook = self.weight.register_hook(
-                lambda grad: grad * keep_mask.type(grad.dtype).to(grad.device)
-            )
-            self.pruning_iterations += 1
+            mu_in = input_tensor.mean()
+            mu_out = output_tensor.mean()
+
+            self.input_means = np.append(self.input_means, mu_in.to("cpu").item())
+            self.output_means = np.append(self.output_means, mu_out.to("cpu").item())
+
+    def progress_connections(self, *args, **kwargs):
+        super().progress_connections(*args, **kwargs)
+        self.coactivations.data[:] = torch.ones_like(self.weight)
 
-            if self.last_keep_mask is not None:
-                kept = self.last_keep_mask == keep_mask
-                kept = kept[keep_mask == 1]
-                self.kept_frac = 1 - calc_sparsity(kept)
-            self.last_keep_mask = keep_mask
 
-    def update_coactivations(self, *args, **kwargs):
+class SparseConv2d(DSConv2d):
+
+    def _init_logging_params(self):
+        pass
+
+    def progress_connections(self, *args, **kwargs):
         pass
 
+    def update_coactivations(self, *args, **kwargs):
+        pass
 
-class SparseConv2d(SparseWeights2d):
-    """
-    Conv layer with static sparsity.
-    """
+# class SparseConv2d(SparseWeights2d):
```

Comment:
Do we really need to keep this commented out code? If not, best to remove it.

Reply:
Good point, I'll remove this and the others above.